lib/operations/command_v2.js

```diff
@@ -15,10 +15,18 @@ class CommandOperationV2 extends OperationBase {
   constructor(parent, options, operationOptions) {
     super(options);
 
+    const noInherit = options && options.noInherit;
+    const inherit = noInherit ? {} : parent;
+
     this.ns = parent.s.namespace.withCollection('$cmd');
-    this.readPreference = resolveReadPreference(parent, this.options);
-    this.readConcern = resolveReadConcern(parent, this.options);
-    this.writeConcern = resolveWriteConcern(parent, this.options);
+    try {
```

Comment:
I think it would be preferable not to add a try/catch to solve this problem, but rather modify `resolveReadPreference` to handle a nullish input for `parent`

Also, FWIW didn't you make `ReadPreference.resolve` ?